api/agent/agent.go

```diff
@@ -726,14 +728,24 @@ func (c *container) Timeout() time.Duration         { return c.timeout }
 func (c *container) EnvVars() map[string]string     { return c.env }
 func (c *container) Memory() uint64                 { return c.memory * 1024 * 1024 } // convert MB
 
-// Log the specified stats to a tracing span.
-// Spans are not processed by the collector until the span ends, so to prevent any delay
-// in processing the stats when the function is long-lived we create a new span for every call
+// WriteStat logs each metric in the specified Stats structure
+// It does this by logging appropriate field values to a tracing span.
 func (c *container) WriteStat(ctx context.Context, stat drivers.Stat) {
+
+	// Spans are not processed by the collector until the span ends, so to prevent any delay
+	// in processing the stats when the function is long-lived we create a new span for every call
 	span, ctx := opentracing.StartSpanFromContext(ctx, "docker_stats")
 	defer span.Finish()
+
 	for key, value := range stat.Metrics {
-		span.LogFields(log.Uint64("fn_"+key, value))
+		// The field name we use is the metric name prepended with "fn_histogram_" to designate that it is a Prometheus histogram metric
+		// The collector will remove "histogram_" and use the result as the Prometheus metric name.
+		fieldname := "fn_histogram_" + "docker_stats_fn_" + key
```

Comment:
we could likely codify this (can shove in common) to make it a little less error prone, as was done with `IncrementGauge`, though every stats interface has its formatting idiosyncrasies:

```go
func Histogram(span opentracing.Span, key string, value int64) {
  span.LogFields(log.Uint64("fn_histogram_"+key, value))
}
```